### ISSUE TITLE
Data doesn't refresh for UDP or WebSocket

### ISSUE THREAD

**Issue by sumocomputers** [bpennypacker/WeatherFlow-Indigo-Plugin#41]
I got my new WeatherFlow Tempest setup using the plug-in version 2022.1.0.

I setup a device for both UDP and WebSocket, and decided to display as the device state, Date/Time last observed.

When the device is first added, it seems to update every minute, but only for a few minutes. Then the device stops updating (I waited hours, data was still stale on both, and date stamp never changed).

I found if I disable and enable device communication, it updates again.

So for now I have a schedule to disable/enable the devices every 30 seconds, and all is good.

I'm not sure if this is a known issue, or if there is some troubleshooting I can do to assist tracking it down.

Thanks,

Chris

**Comment by 4ParMan** [comment]
I have a similar experience. None of my UDP data updates in Indigo, and some of my websockets does not update.
When having indigo server off, I can see the incoming data in terminal, b ut no update tin indigo.

**Comment by sumocomputers** [comment]
@bpennypacker Just curious if you are maintaining this plugin any more?

**Comment by freshwuzhere** [comment]
On my setup UDP or web socket work great for one (the latest) variable unfortunately it stops any other variable from updating.  This is similar for UDP direct or Websocket etc so maybe a code bug.   Seems like plugin isn't being updated - so may have to see about debugging myself........
Anyone else already fixed this?

Oh I have a fix if anyone is interested - Only one device the latest you created will update - but the states in that object are correct so you can copy them to a variable with a trigger (or use them for whatever you want) and the data appears to match the latest update from any source.

**Comment by epasc** [comment]
@freshwuzhere does your workaround still work?

I have tried UDP and websockets. The device states populate when the device is created but never update again. They just remain constant. But maybe I’m doing something wrong.

I am not a coder or developer so I don’t know the next step, but can anyone recommend a way to get this working (maybe by hiring a python developer). It’s important to me to get this working reliably

**Comment by freshwuzhere** [comment]
@epasc  YES - just checked it and working as planned - trigger fired by "any change" works as expected.

Still running indigo 2024.2.0 *haven't upgraded to 2024.2.1 yet and you are making me nervous!!  

   Trigger                         Set Wind Lull
   Trigger                         Set Wind Avg
   Trigger                         Set Wind Gust
   Trigger                         Set Wind Direction
   Trigger                         Set OutsideT

**Comment by epasc** [comment]
Sadly it does not work on my system which is 2024.2.1

**Comment by hamwi875** [comment]
Any updates on this? Would be great if it would work. If not, perhaps it should be noted as an unsupported plugin, as I just spent an hour or so trying to get it to work. Anyone have success with Home Assistant? They have a device which I set up, but not sure how to read into an indigo Home Assistant device.

**Comment by plorimer** [comment]
I'm on Indigo 2024.2.1 and WeatherFlow v2022.0.1 is working fine for me.

<img width="464" height="322" alt="Image" src="https://github.com/user-attachments/assets/b7467e4b-239c-4a11-a616-111764128aeb" />

**Comment by hamwi875** [comment]
I wonder what I'm doing wrong… My Indigo instance can see the Tempest itself, and the terminal and indigo logging show that it is working, at least on UDP. However, the tempest device fields don't populate.

I was able to work around this by creating separate Tempest devices for each of temperature, humidity, and precipitation, and they work. Maybe I just need to give it some time to populate.

**Comment by plorimer** [comment]
What model do you have it set to? I'm using the Tempest Websocket Device: 

<img width="603" height="620" alt="Image" src="https://github.com/user-attachments/assets/8fd881a4-ab16-47d3-ab3a-354b4296910a" />

**Comment by hamwi875** [comment]
That's how I tried, but it wouldn't populate. Away right now and will try again in a little bit.  FWIW am on 2025.1.

**Comment by bpennypacker** [comment]
Unfortunately I no longer have time to maintain this. I switched from Indigo to Home Assistant when we moved, and I'm using that for all my home automation now. Home Assistant actually has good support for both [local](https://www.home-assistant.io/integrations/weatherflow/) & [cloud](https://www.home-assistant.io/integrations/weatherflow_cloud) based access to a Tempest device. Those are what I'm currently using.

If anybody is interested in taking over ownership of this plugin for Indigo I'd be more than happy to do that.

**Comment by hamwi875** [comment]
<img width="885" height="463" alt="Image" src="https://github.com/user-attachments/assets/b615567c-91c5-49c5-ab2e-90adc104a553" />
Hi guys, thanks for the prompt responses. I got the devices in via Home assistant plugin as the WeatherFlow plugin was not initially working. Those are the individual temp, hum, pressure, etc. Glad to say that when I got back home the plugin is definitely working and populating as intended. So it works in 2025.1. Bruce, thanks for your effort in creating this plugin, and good luck with HA.

**Comment by hamwi875** [comment]
I think I may have found the issue. In the setup, it advises to go to the map, highlight the station and get what appears to be a 4 digit id code. I did that, finding a 6 digit code, and entering the last 4, my guess as to what the proper 4 digits would be. However, the unit would not read. I then put in the full 6 digit code and it worked. Hope that helps. Must be a new revision or perhaps they have just sold more units, so now 6 digits is correct.
